sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala

```diff
@@ -408,10 +408,12 @@ private[hive] class HiveClientImpl(
     Option(shim.getDatabase(client, dbName)).map { d =>
       val params = Option(d.getParameters).map(_.asScala.toMap).getOrElse(Map()) ++
         Map(PROP_OWNER -> shim.getDatabaseOwnerName(d))
+      val collation = params.get(PROP_COLLATION)
 
       CatalogDatabase(
         name = d.getName,
         description = Option(d.getDescription).getOrElse(""),
+        collation = collation,
```

Comment:
It seems to me that this new field only brings troubles, not benefits. Hive database does not have this field, so we need to turn it into a property anyway. We only have one place to access this field, and we can simply get collation property and not have this field.